metrics-operator/controllers/analysis/controller.go

```diff
@@ -78,12 +78,14 @@ func (a *AnalysisReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	analysis.EnsureTimeframeIsSet()
```

Comment:
Is this actually needed if we call `EnsureTimeFrameSet` always as part of `GetFrom` or `GetTo` functions ?

Reply:
Or vice versa, if it's set in here, we can remove it from these functions, or am I missing some info ?

Reply:
you're right - we can remove it from the `GetTo/GetFrom` functions. I was moving that part back and forth as I could not decide where exactly to put it, but in the end I believe it has to be the controller who calls that function, as the controller is responsible for handling and setting the state of the analysis

Reply:
agree